src/OpsGenieNotification.groovy

```diff
@@ -109,27 +110,43 @@ def sendAlert(Map executionData, Map configuration) {
     }
 
     // Send the request.
-    def url = new URL(DEFAULTS.OPSGENIE_URL)
+    def endpoint = DEFAULTS.OPSGENIE_URL;
+    if (close) {
+        // TODO: Alias needs url escaping
+        def encodedAlias = java.net.URLEncoder.encode(expandedAlias, "UTF-8")
+        endpoint += "/${encodedAlias}/close?identifierType=alias"
+        // TODO: Add auto-close note.
+    }
+    def url = new URL(endpoint)
     def connection = url.openConnection()
     connection.setRequestMethod("POST")
-    connection.addRequestProperty("Content-type", "application/json")
+    connection.setRequestProperty("Content-Type", "application/json")
+    connection.setRequestProperty("Authorization", "GenieKey ${configuration.api_key}")
+    connection.setRequestProperty("Accept", "application/json")
     connection.doOutput = true
+
     def writer = new OutputStreamWriter(connection.outputStream)
     def json = new ObjectMapper()
-    def raw_data = json.writeValueAsString(job_data)
+    String raw_data = null
+    if (close) {
+        raw_data = json.writeValueAsString([note: "Auto closing as job has since succeeded"])
+    } else {
+        raw_data = json.writeValueAsString(job_data)
+    }
     System.err.println("DEBUG: request: " + raw_data)
     writer.write(raw_data)
     writer.flush()
     writer.close()
     connection.connect()
 
     // process the response.
-    def response = connection.content.text
-    System.err.println("DEBUG: response: " + response)
-    JsonNode jsnode = json.readTree(response)
-    def status = jsnode.get("status").asText()
-    if (!"success".equals(status)) {
-        System.err.println("ERROR: OpsGenieNotification plugin status: " + status)
+    int response_code = connection.getResponseCode()
+    if (response_code != 202) {
```

Comment:
I guess anything with <300 should be valid

Reply:
Possibly but the OpsGenie documentation does say that a 202 will be returned. This is also the same for the Close Alert endpoint.

"Create alert requests are processed asynchronously, therefore valid requests are responded with HTTP status 202 - Accepted."

https://docs.opsgenie.com/docs/alert-api#section-create-alert